.github/workflows/release.yml

```diff
@@ -77,3 +77,23 @@ jobs:
           GIT_AUTHOR_EMAIL: ${{ vars.BOT_GIT_AUTHOR_EMAIL }}
           GIT_COMMITTER_NAME: ${{ vars.BOT_GIT_COMMITTER_NAME }}
           GIT_COMMITTER_EMAIL: ${{ vars.BOT_GIT_COMMITTER_EMAIL }}
+
+      - name: Setup cmake
+        uses: jwlawson/actions-setup-cmake@v2
+        with:
+          cmake-version: "3.29.0"
+
+      - name: Publish crates to crates.io
```

Comment:
The publication should be part of its own workspace I guess, to separate the release from the publish. And workflow could be triggered by the addition of a new tag.